npc/jobs/2-2/crusader.txt

```diff
@@ -223,6 +209,27 @@ prt_castle,45,169,5	script	Senior Crusader	4_M_CRU_OLD,{
 	else if(CRUS_Q >= 1 && CRUS_Q <= 3) {
 		mes "Have you proven your determination with the task I have given you, or do you possess the items proving that you have received your calling?";
 		next;
+		if (countitem(Patriotism_Marks) >= 1 && countitem(Sacred_Marks) >= 1) {
+			mes "Ah...";
+			mes "I see that you have been called to become a Crusader. We are assured of your will, but now we must test your capabilities.";
+			next;
+			mes "[Michael Halig]";
+			mes "Meet with Moorenak Miyol who is training in the underground dungeon of the Prontera Castle. Go, and speak with him first.";
+			next;
+			delitem Patriotism_Marks, 1;
+			delitem Sacred_Marks, 1;
+			CRUS_Q = 4;
+			if (questprogress(3006)) {
+				changequest 3006,3009;
+			} else if (questprogress(3007)) {
+				changequest 3007,3009;
+			} else {
+				changequest 3008,3009;
```

Comment:
I am no fan of changequest in such cases, simply because the quest progress is supposed to be 2 (completed) afterwards, but with changequest it got resetted (deleted) and a future check wouldn't be possible. 
With proper usage of questprogress, comoletequest and erasequest we could avoid so many additional variables.